src/main/java/com/iexec/worker/pubsub/StompClient.java

```diff
@@ -142,26 +150,42 @@ private void requestNewSession() {
      * different calls (and possibly different threads), then we send
      * only one request to the server. This process is repeated until
      * the websocket connection is reestablished again.
-     * 
-     * @throws InterruptedException
      */
-    @Scheduled(fixedDelay = 1000)
-    private void listenToSessionRequestEventsInTheQueue() throws InterruptedException {
-        while (true) {
+    void listenToSessionRequests() {
+        log.info("Listening to incoming STOMP session requests");
+        while (!Thread.interrupted()) {
             // get the first request event or wait until available
-            this.sessionRequestQueue.take();
+            try {
+                this.sessionRequestQueue.take();
+            } catch (InterruptedException e) {
+                log.error("Interrupted while listening to incoming STOMP session requests", e);
+                Thread.currentThread().interrupt();
+                return;
+            }
             // wait some time for the wave of request events coming
             // from possibly different threads to finish
-            TimeUnit.SECONDS.sleep(SESSION_REFRESH_DELAY);
-            log.info("Creating new STOMP session");
+            try {
+                TimeUnit.SECONDS.sleep(SESSION_REFRESH_DELAY);
+            } catch (InterruptedException e) {
+                log.error("Interrupted while sleeping", e);
+                Thread.currentThread().interrupt();
+                return;
+            }
             // purge redundant request events
             this.sessionRequestQueue.clear();
             // Only one attempt should pass through
-            log.debug("Sending new STOMP connection request");
-            this.stompClient.connect(webSocketServerUrl, new SessionHandler());
+            createSession();
         }
     }
```

Comment:
What if the listener crashes? In the previous version, it was restarted by Spring. Now, it doesn't look to do so.

Reply:
See [this](https://github.com/iExecBlockchainComputing/iexec-worker/pull/402#discussion_r731929241) comment.